crates/edr_eth/src/spec.rs

```diff
@@ -26,10 +26,10 @@ pub trait ChainSpec {
 }
 
 /// Constants for constructing Ethereum headers.
-pub trait EthHeaderConstants: ChainSpec<Hardfork: 'static + PartialOrd> {
-    /// Parameters for the EIP-1559 base fee calculation.
-    const BASE_FEE_PARAMS: BaseFeeParams<Self::Hardfork>;
```

Comment:
I think it would be clearer to the user that the `fn chain_base_fee_params` can be wrong if we return an `Option`. That way we are signalling: "If the chain_id is known, we will return its BaseFeeParams".

That would require the a default `BASE_FEE_PARAMS` to remain. Maybe just rename it to `DEFAULT_BASE_FEE_PARAMS`.

At a call site, the user can decide to do:

```rs
// reverting example
ChainSpecT::chain_base_fee_params(chain_id).or(ChainSpecT::DEFAULT_BASE_FEE_PARAM)

// propagation error
ChainSpecT::chain_base_fee_params(chain_id).ok_or_else(|| anyhow!("Chain ID is not supported"))?;
```